### ISSUE TITLE
postresolve function never processes rpz triggered requests

### ISSUE THREAD

**Issue by thinkwelltwd** [PowerDNS/pdns#9370]
- Program: Recursor
 - Issue type: Bug report

### Short description
postresolve lua hook never gets DNS Questions that were found in a response policy zone

### Environment
 - Operating system: Centos  6 & 7
 - Software version: 4.3.3
 - Software source: PowerDNS repository

### Steps to reproduce

recursorconf.lua

```
rpzFile("/path/to/zone.conf", {defpol=Policy.Custom, defcontent="urlhaus.localblockpage.lan"})
```

lua_dns_script.lua

```lua
function prerpz(dq)
  pdnslog(dq.qname:toString() .. ' prerpz')
  return false
end


function postresolve(dq)
  pdnslog(dq.qname:toString() .. " postresolve" )
  return false
end
```

### Expected behaviour

Log line when RPZ fired

```
Jul 31 19:48:39 system pdns_recursor: aite.me. prerpz
Jul 31 19:48:39 system pdns_recursor: aite.me. postresolve
```

Log lines of when no RPZ fired
```
Jul 31 19:49:11 system pdns_recursor: lxer.com. prerpz
Jul 31 19:49:11 system pdns_recursor: lxer.com. postresolve
```

### Actual behaviour
There's no postresolve logline for the domains in the urlhaus RPZ, so no way of logging malware activity.

Log line when RPZ fired
```
Jul 31 19:48:39 system pdns_recursor: aite.me. prerpz
```

Log lines of when no RPZ fired
```
Jul 31 19:49:11 system pdns_recursor: lxer.com. prerpz
Jul 31 19:49:11 system pdns_recursor: lxer.com. postresolve
```

### Question
Presumably this would be the approach to logging RPZ hits to a file for further processing. I also tried `dnstap` config but wasn't able to get that working. In reading [this issue](https://github.com/PowerDNS/pdns/issues/6373) it doesn't seem like dnstap in Recursor is ready for prime time.

**Comment by phonedph1** [comment]
If you just need to do something to log them - the policy being applied is available in `preresolve` and you can just check `dq.appliedPolicy.policyName ~= ""` or similar.

**Comment by thinkwelltwd** [comment]
Indeed that is right; `preresolve` gets me what I want. I'd read over the hooks carefully and wondered what the `postrpz` hook was. Apparently it's `preresolve` and I suppose I should've intuited that.

If the docs could be changed from:

> This function is called before any DNS resolution is attempted, 

to:

> This function is called after RPZs are applied and  before any DNS resolution is attempted, 

I'll leave the ticket open because the behaviour I'm seeing does not match the docs. `postresolve` should really be called...

Anyway, this does get me logging, so very many thanks @phonedph1!

**Comment by rgacogne** [comment]
> `postresolve` should really be called...

I agree the documentation should be more explicit on that, but I don't think it really makes sense to call `postresolve` in that case since the RPZ hit prevents the resolution process from even starting.
